next-env.d.ts

```diff
@@ -1,5 +1,6 @@
 /// <reference types="next" />
 /// <reference types="next/image-types/global" />
+/// <reference types="next/navigation-types/compat/navigation" />
```

Comment:
It seems like this got added when I made this file folder `app/api/auth/[...nextauth]/route.ts`. The reason why I need this folder file is because the new auth.js version is switching over to app router for next.js it only works if I have the app folder setup like this for auth.js v5. link for reference: https://authjs.dev/getting-started/migrating-to-v5. it is under 1. Configuration File.

Reply:
It is typescript types for navigating through routes. link: https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating